db.go

```diff
@@ -198,34 +290,40 @@ func (x *sanitizingAuthenticator) Authenticate(identity, password string) (strin
 	return x.backend.Authenticate(identity, password)
 }
 
-func (x *sanitizingAuthenticator) SetPassword(identity, password string) error {
-	identity, password = cleanIdentityPassword(identity, password)
-	if len(identity) == 0 {
+func (x *sanitizingAuthenticator) Close() {
+	if x.backend != nil {
+		x.backend.Close()
+		x.backend = nil
+	}
+}
+
+func (x *sanitizingUserStore) SetPassword(userId UserId, password string) error {
+	password = cleanPassword(password)
+	if userId <= 0 {
```

Comment:
How can we know that userID <= 0 is an invalid userid?
Also, your original UserId was uint64, so testing for less than zero is pointless. If we absolutely need to, then I think it's OK to say that userID = 0 is an invalid userID, because I know of no databases that start generating autoincrementing IDs at zero (they all start at 1). However, don't make any such restriction unless it's necessary. In this case, there is no need. x.backend.SetPassword() will simply tell us that the ID is unknown, if that is the case.